hackathon_site/hardware/test_api.py

```diff
@@ -1258,3 +1258,70 @@ def test_no_remaining_quantities(self):
             ],
         }
         self.assertEqual(response.json(), expected_response)
+
+
+class OrderListPatchTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.view_name = "api:hardware:order-detail"
+        self.change_permissions = Permission.objects.filter(
+            content_type__app_label="hardware", codename="change_order"
+        )
+        hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+        order = Order.objects.create(status="Submitted", team=self.team)
+        OrderItem.objects.create(order=order, hardware=hardware)
+        self.pk = order.id
+
+    def _build_view(self, pk):
+        return reverse(self.view_name, kwargs={"pk": pk})
+
+    def test_user_not_logged_in(self):
+        request_data = {"status": "Ready for Pickup"}
+        response = self.client.patch(self._build_view(self.pk), request_data)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_lack_perms(self):
+        self._login()
+        request_data = {"status": "Ready for Pickup"}
+        response = self.client.patch(self._build_view(self.pk), request_data)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_successful_status_change(self):
+        self._login(self.change_permissions)
+        request_data = {"status": "Ready for Pickup"}
+        response = self.client.patch(self._build_view(self.pk), request_data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(request_data["status"], Order.objects.get(id=self.pk).status)
```

Comment:
This still doesn't guarantee that the field was actually changed - if the serializer discarded the field, this test would still pass. You'll need to test that the object's status is explicitly "Ready for Pickup". Same for the other tests.

Reply:
I'm not sure I understand. I thought I was checking whether the object's status (Order.objects.get(id=self.pk).status)) is explicitly "Ready for Pickup" (request_data["status"])

Reply:
are you suggesting if somehow, it took out the "status" field from request_data, in the self.client.patch?

Reply:
I am not doing very good with PRs lately, I was reading that as response["status"] 🤦‍♂️. You're totally right, this is perfect as-is!